src/CoreFx.Private.TestUtilities/src/System/IO/FileCleanupTestBase.cs

```diff
@@ -76,6 +76,14 @@ protected virtual void Dispose(bool disposing)
         /// </summary>
         protected string TestDirectory { get; }
 
+        /// <summary>Creates a test directory that is associated with the call site.</summary>
+        protected string CreateTestDirectory(int? index = null, [CallerMemberName] string memberName = null, [CallerLineNumber] int lineNumber = 0)
```

Comment:
I'm almost sure that xunit will create one instance of class per test which means you should be able to just use TestDirectory. Are you hitting any issues with that? The class representing test directory with additional member which also creates directory and is not tracked by this class seems a bit weird

Reply:
after skimming through the tests I'm wondering if this directory should be a subdirectory of TestDirectory so that it gets auto-deleted

Reply:
> after skimming through the tests I'm wondering if this directory should be a subdirectory of TestDirectory

That is the case.

> if possible to not have this method please do it

I'll remove it.